### PR TITLE
[rush] Add new command "rush purge"

### PR DESCRIPTION
**Pull request by pgonzal**
This adds a "rush purge" command for cleaning up temporary files.  The upcoming "rush update" PR will simplify the installation algorithm by no longer intermixing purging with installation.

### CODE REVIEW

File: apps/rush-lib/src/utilities/AsyncRecycler.ts [hunk 1]
```diff
@@ -77,6 +78,40 @@ export class AsyncRecycler {
...
+      if (!excludeSet.has(normalizedMemberName)) {
+        let shouldMove: boolean = false;
+        try {
+          const stats: fsx.Stats = fsx.lstatSync(memberPath);
+          shouldMove = stats.isDirectory();
+        } catch (error) {
+          // If we fail to access the item, assume it's not a folder
+        }
+        if (shouldMove) {
+console.log('==> Recycling: ' + memberPath);
```
Comment (iclanton) [inline apps/rush-lib/src/utilities/AsyncRecycler.ts]: >console [](start = 0, length = 7)

What's with the weird indentation?

File: apps/rush-lib/src/cli/logic/PurgeManager.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+   */
+  public purgeNormal(): void {
+    // Delete everything under common\temp except for the recycler folder itself
+    console.log('Purging ' + this._rushConfiguration.commonTempFolder);
+    this._commonTempFolderRecycler.moveAllItemsInFolder(this._rushConfiguration.commonTempFolder,
+      [RushConstants.rushRecyclerFolderName]);
+  }
+
+  /** */
+  public purgeUnsafe(): void {
```
Comment (iclanton) [inline apps/rush-lib/src/cli/logic/PurgeManager.ts]: Docs?

Comment (iclanton) [inline apps/rush-lib/src/cli/logic/PurgeManager.ts]: >purgeUnsafe [](start = 9, length = 11)

Make sure you don't delete the folder that rush is executing in.

File: apps/rush-lib/src/cli/logic/InstallManager.ts [hunk 3]
```diff
@@ -170,7 +170,7 @@ export class InstallManager {
       packageManagerVersion: rushConfiguration.packageManagerToolVersion
     });
 
-    this._asyncRecycler = new AsyncRecycler(this._rushConfiguration);
+    this._asyncRecycler = new AsyncRecycler(this._rushConfiguration.commonTempFolder);
```
Comment (iclanton) [inline apps/rush-lib/src/cli/logic/InstallManager.ts]: >commonTempFolder [](start = 68, length = 16)

This should have `RushConstants.rushRecyclerFolderName` appended to it.

File: apps/rush-lib/src/utilities/AsyncRecycler.ts (hunk 1 above)
Comment (pgonzal) [reply apps/rush-lib/src/utilities/AsyncRecycler.ts]: That was a reminder to delete this debug code.  :-)

Comment (pgonzal) [reply apps/rush-lib/src/utilities/AsyncRecycler.ts]: That was a reminder to delete this debug code.  :-)

---
In reply to: [186247303](https://github.com/Microsoft/web-build-tools/pull/643#discussion_r186247303) [](ancestors = 186247303)

**Pull request merged**